crawl-ref/source/dat/des/branches/lair.des

```diff
@@ -2748,6 +2778,107 @@ MAP
   -------------------
 ENDMAP
 
+NAME:   pdpol_lair_dawg_pound
+TAGS:   no_monster_gen
+DEPTH:  Lair
+KMONS:  1 = wolf / warg w:5
+KMONS:  2 = hell hound / raiju
+KITEM:  2 = *
+SUBST:  _ = x _:3, S = S:1 . %:2
+NSUBST: s = 1:S / *= . S:1, D = 2:2 / 4= 1 2:1 / *= . 1:2
+: dgn.delayed_decay(_G, 'S', "felid skeleton")
+CLEAR:  _
+MAP
+            ________
+          __xxxxxxx__
+          _xx.SS..x__
+      _____x..SS..x__
+    ___xxxxx....DDx__
+ ____xxx..SS..DDxxx__
+ _xxxx.....SxDxxxxx__
+_xxSS....x..xxxx____
+_xDSS..Dxxssxxxx__
+_xS..Dxxxssxx____
+_xxDxxxxssxxx_
+__xxxxx..xx_
+    xx@@xx
+ENDMAP
+
+# Dead by dawn!
+NAME:    pdpol_lair_cabin_in_the_woods
+TAGS:    no_item_gen no_monster_gen
+DEPTH:   Lair
+# Trying to get some good horror move trope representation
+KMONS:   M = Rupert / reaper / vampire knight / phantasmal warrior / tentacled starspawn
```

Comment:
This will place no monster if it rolls Rupert and he's already generated elsewhere. You can do M = Rupert, ghoul / reaper / etc to place a backup monster if he has already spawned (feel free to use a different monster of course). I'd also recommend using a vampire mage over a vampire knight, since that one stands out as more difficult than the other non-Rupert options imo.

Reply:
Oh whoops that's good to know. So with that comma-separated syntax, will it always try to place Rupert first? Or would there still be a chance to place the other monsters even if Rupert wasn't placed elsewhere? Could I do something like `M = Rupert / ghoul / reaper / etc, ghoul / reaper / etc` so that there's an equal chance but then it handles the Rupert picked but unavailable case?

Reply:
Also will go ahead and change that vampire knight.

Reply:
I think it actually should use a glyph to subst Rupert or his backup in with some chance and then do the rest of the kmons line normally. There's probably some one line syntax that works but I don't remember what it is, you'd have to test it. Definitely doesn't work the way I originally wrote it though, sorry about that.

Reply:
I think I managed to find a way to do this just building `kmons` in a lua chunk with `you.uniques("Rupert")` to determine whether to throw him in or not, but let me know if that isn't the best way to do it.